shpc/main/modules/base.py

```diff
@@ -475,8 +456,47 @@ def install(
             name = "%s:%s" % (name, tag.name)
         logger.info("Module %s was created." % name)
 
-        # Install the module (symlink) to the view and create version file
-        if view:
-            view.install(module_dir)
-
         return container_path
+
+    def view_install(self, view_name, name, tag=None, force=False):
+        """
+        Install a module in a view.
+        The module must already be installed.
```

Comment:
Perhaps the repeated logic at the beginning of both functions could be in a common load module function? And it should return some class that transparently provides all the needed variables (that aren’t in the config, which would be one of the variables).

Reply:
Possibly, yes. It sounds like you've got something in mind, so I'll let you play around

Reply:
In case you missed it! https://github.com/muffato/singularity-hpc/pull/4#issue-1403058615

Reply:
I had missed it :grin: Thanks !

Reply:
Yeah I woke up in the middle of the night like some kind of goblin... just in time for Halloween! :japanese_goblin: